From 42521c696dea653657bd8b2fa05e26a1ea157765 Mon Sep 17 00:00:00 2001
From: Christina Jacob <cjacob@marvell.com>
Date: Tue, 10 Sep 2019 08:59:48 +0530
Subject: [PATCH 04/15] octeontx2-pf: remove redundant changes from speed
 change suppcrt.

commit a080c57ef72e94b8de3a3abeea76692f78050dc5 from
git@git.assembla.com:cavium/WindRiver.linux.git

clean up speed change support via ethtool.

Change-Id: I3c9f606f04c67c01421660bb3759331aa5fd80b0
Signed-off-by: Christina Jacob <cjacob@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15605
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../ethernet/marvell/octeontx2/nic/otx2_ethtool.c   | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index eec6c13122bb..80aaf80d3951 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -919,14 +919,6 @@ static void otx2_get_fec_info(u64 index, int mode, struct ethtool_link_ksettings
 	}
 }
 
-static u32 otx2_get_link_mode_info(u64 index)
-{
-	int link_modes = 0;
-
-	/************  TO DO *********/
-	return link_modes;
-}
-
 static struct cgx_fw_data *otx2_get_fwdata(struct otx2_nic *pfvf)
 {
 	struct cgx_fw_data *rsp = NULL;
@@ -1001,13 +993,10 @@ static int otx2_get_link_ksettings(struct net_device *netdev,
 
 	if (rsp->fwdata.supported_an)
 		supported |= SUPPORTED_Autoneg;
-	advertising |= otx2_get_link_mode_info
-			(rsp->fwdata.advertised_link_modes);
 	ethtool_convert_legacy_u32_to_link_mode(cmd->link_modes.advertising,
 						advertising);
 	otx2_get_fec_info(rsp->fwdata.advertised_fec, 1, cmd);
 
-	supported |= otx2_get_link_mode_info(rsp->fwdata.supported_link_modes);
 	ethtool_convert_legacy_u32_to_link_mode(cmd->link_modes.supported,
 						supported);
 	otx2_get_fec_info(rsp->fwdata.supported_fec, 0, cmd);
@@ -1040,8 +1029,6 @@ static int otx2_set_link_ksettings(struct net_device *netdev,
 	else
 		req->args.duplex = cmd->base.duplex ^ 0x1;
 	req->args.an =  cmd->base.autoneg;
-	req->args.ports = cmd->base.port;
-	req->args.flags = 0;
 
 	err =  otx2_sync_mbox_msg(&pfvf->mbox);
 	if (!err) {
-- 
2.17.1

